TelCo.ColorCoder/ColorMap.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Diagnostics;
+using System.Drawing;
+
+namespace TelCo.ColorCoder
+{
+        /// <summary>
+        /// data type defined to hold the two colors of clor pair
+        /// </summary>
+        internal class ColorPair
```

Comment:
you have created good modular files, did you not think of moving ColorPair to a new file?

Reply:
I thought overall color related data placed in one file. It makes sense to place each class in separate file as each class providing different functionality.